web/impact/impact/v1/views/office_hours_calendar_view.py

```diff
@@ -323,23 +330,28 @@ def _make_f_expression(field, location_path):
     return F("{}__{}".format(location_path, field))
 
 
-def _date_range(date_spec=None):
+def _date_range(calendar_span, focal_date=None):
     # returns (start_date, end_date)
     # start_date is the latest monday that is less than or equal to today,
     # end_date is start_date + seven days
     # both values are then padded by 24 hours to allow for TZ differences
 
-    # throws ValueError if date_spec is not in ISO-8601 format
+    # throws ValueError if focal_date is not in ISO-8601 format
 
-    if date_spec:
-        initial_date = datetime.strptime(date_spec, ISO_8601_DATE_FORMAT)
+    if focal_date:
+        initial_date = datetime.strptime(focal_date, ISO_8601_DATE_FORMAT)
     else:
         initial_date = datetime.now()
     initial_date = utc.localize(initial_date)
     # This calculation depends on the fact that monday == 0 in python
-    start_date = initial_date - timedelta(initial_date.weekday())
-    end_date = start_date + ONE_WEEK + ONE_DAY
-    adjusted_start_date = start_date - ONE_DAY
+    if calendar_span == "monthly":
+        start_date = initial_date - timedelta(int(initial_date.strftime("%d")))
+        end_date = start_date + ONE_MONTH + ONE_DAY
+        adjusted_start_date = start_date + ONE_DAY
```

Comment:
shouldn't this be minus ONE_DAY?

Reply:
Nope

Reply:
```
>>> initial_date = datetime.now()
>>> start_date = initial_date - timedelta(int(initial_date.strftime("%d")))
>>> start_date
datetime.datetime(2020, 8, 31, 8, 59, 11, 83622)
>>> start_date + timedelta(1)
datetime.datetime(2020, 9, 1, 8, 59, 11, 83622)
```

Reply:
I am not sure I follow

Reply:
9/1/2020 at 8:59 is nine hours into the month. When we use this to filter, we will miss any sessions scheduled for the morning of 9/1 (in this case)
```
        return MentorProgramOfficeHour.objects.filter(
            active_mentors,
            start_date_time__range=[self.start_date, self.end_date])
```

If we just take the originally-calculated initial_date, we get a date that has effectively got an adjustment between 1 and 3600 seconds before the start of month, which is good, but leaves us maybe vulnerable to timezone damage. ie, there may be scenarios where a user at certain times-of-day in certain timezones might end up not getting all of the sessions scheduled for that month, in their timezone. 

The easiest solution is to simply add 24 hours on either end as a buffer, which is what we do in the week case. This ensures that the front-end gets everything it needs (and if it gets some extra, it can drop those on the floor, we don't care)

Reply:
Thanks for clarifying